### PR TITLE
feat(lyrics-plus): add netease translation

### PR DESCRIPTION
**Pull request by lixiao189**
image

Add Chinese translation option when this song has netease translation.

Maybe it can solve this issue: https://github.com/spicetify/spicetify-cli/issues/1411#issue-1104981280

### CODE REVIEW

**Comment by lixiao189** [comment]
@theRealPadster @afonsojramos Hello, nobody replied here. Can you review this PR, please? 😘

**Comment by theRealPadster** [comment]
Sorry, I don't know how this custom app works so don't think a review would be particularly helpful

**Comment by lixiao189** [comment]
> Sorry, I don't know how this custom app works so don't think a review would be particularly helpful

Thank you for reply. I found that this custom app was maintained by a Chinese developer, so what is his username? Mayby I can mention him to review this code.

**Comment by kyrie25** [comment]
> > Sorry, I don't know how this custom app works so don't think a review would be particularly helpful
> 
> Thank you for reply. I found that this custom app was maintained by a Chinese developer, so what is his username? Mayby I can mention him to review this code.

Developed by `@khanhas` and maintained by the team, not exactly one/any dev alone. I'll look into it sometime within this weekend, unless somebody else beats me to it.

**Comment by lixiao189** [comment]
> > > Sorry, I don't know how this custom app works so don't think a review would be particularly helpful
> > 
> > 
> > Thank you for reply. I found that this custom app was maintained by a Chinese developer, so what is his username? Mayby I can mention him to review this code.
> 
> Maintained by the team, not exactly one/any dev alone. I'll look into it sometime within this weekend, unless somebody else on the team beats me to it.

Ok, thank you

File: CustomApps/lyrics-plus/ProviderNetease.js [hunk 1]
```diff
@@ -147,6 +147,37 @@ const ProviderNetease = (function () {
 		return lyrics;
 	}
 
+	function getTranslation(list) {
+		const lyricStr = list?.tlyric?.lyric;
```
Comment (kyrie25) [inline CustomApps/lyrics-plus/ProviderNetease.js:151]: Does this only gets Chinese translations? Or any translations there is of the song.

File: CustomApps/lyrics-plus/index.js [hunk 2]
```diff
@@ -607,10 +608,9 @@ class LyricsContainer extends react.Component {
 		}
 
 		this.state.mode = mode;
-		const showTranslationButton =
-			(this.state.synced || this.state.unsynced) &&
-			Utils.isJapanese(this.state.synced || this.state.unsynced) &&
-			(mode == SYNCED || mode == UNSYNCED);
+		const hasNeteaseTranslation = this.state.neteaseTranslation !== null;
+		const isJapanese = (this.state.synced || this.state.unsynced) && Utils.isJapanese(this.state.synced || this.state.unsynced);
+		const showTranslationButton = (isJapanese || hasNeteaseTranslation) && (mode == SYNCED || mode == UNSYNCED);
```
Comment (kyrie25) [inline CustomApps/lyrics-plus/index.js:613]: We really need to refactor this some time later lol

File: CustomApps/lyrics-plus/ProviderNetease.js (hunk 1 above)
Comment (lixiao189) [reply CustomApps/lyrics-plus/ProviderNetease.js:151]: Only Chinese translation provide by netease music website

Comment (kyrie25) [reply CustomApps/lyrics-plus/ProviderNetease.js:151]: Fair enough ig, the issue only showcased Netease music displaying Chinese translations

**Review by kyrie25 — APPROVED** [review]

File: CustomApps/lyrics-plus/ProviderNetease.js (hunk 1 above)
Comment (lixiao189) [reply CustomApps/lyrics-plus/ProviderNetease.js:151]: Thank you, sir!

**Comment by lixiao189** [comment]
@theRealPadster [kyrie25](https://github.com/kyrie25) has reviewd and approved these changes.

**Comment by kyrie25** [comment]
> @theRealPadster [kyrie25](https://github.com/kyrie25) has reviewd and approved these changes.

Please don't bother other members of the team. He's not the only reviewer left and this PR can wait.

**Comment by lixiao189** [comment]
> > @theRealPadster [kyrie25](https://github.com/kyrie25) has reviewd and approved these changes.
> 
> Please don't bother other members of the team. He's not the only reviewer left and this PR can wait.

I'm sorry 🙇

**Review by rxri — APPROVED** [review]

**Pull request merged**